Geometry2d/Point.hpp

```diff
@@ -40,6 +42,11 @@ class Point {
     Point(const QPoint& other) : Point(other.x(), other.y()) {}
 
     /**
+     * Implicit constructor for creating a Point from a float*
```

Comment:
Actually never mind, this constructor is still necessary for the k-d tree. There are places where it converts between T (in this case our Geometry2d::Point) and double arrays. I also fixed the comment